e2e/cluster.go

```diff
@@ -133,7 +131,20 @@ func createCluster(ctx context.Context, t *testing.T, cluster *armcontainerservi
 	}
 
 	return &Cluster{Model: createdCluster, Kube: kube, SubnetID: subnetID, NodeBootstrappingConfiguration: nbc}, nil
+}
 
+func hash(cluster *armcontainerservice.ManagedCluster) string {
+	jsonData, err := json.Marshal(cluster)
+	if err != nil {
+		panic(err)
+	}
+	hasher := sha256.New()
+	_, err = hasher.Write(jsonData)
+	if err != nil {
+		panic(err)
+	}
+	hexHash := hex.EncodeToString(hasher.Sum(nil))
+	return hexHash[:5]
```

Comment:
I'm assuming we're only taking 0-4 here to conform with name length constraints?

Reply:
I think we can use up to 64 characters for a cluster name. Potentially can throw another 20 or so.
The main goal is to have enough uniqueness. Something similar to pod names.